File: Java/Hussain Sir Java/Applications/Desktop/SwingMVCTest/src/tour.app/VisitorModel.java
package edu.met.tourism;

public class VisitorModel extends javax.swing.table.AbstractTableModel{

	private Document<Visitor> store = Document.open(Visitor.class, "site.dat");

	@Override
	public int getColumnCount(){
		return 3;
	}

	@Override
	public String getColumnName(int i){
		String[] names = {"Name", "Frequency", "Recent"};
		return names[i];
	}

	@Override
	public int getRowCount(){
		return store.size();
	}

	@Override
	public Object getValueAt(int i, int j){
		Visitor visitor = store.get(i);
		switch(j){
			case 0: return visitor.name;
			case 1: return visitor.frequency;
			default: return visitor.recent;
		}
	}

	public void writeVisitor(String id){
		Visitor visitor = store.find(e -> e.name.equals(id));
		if(visitor == null)
			store.add(new Visitor(id));
		else
			visitor.revisit();
		store.save();
		fireTableDataChanged();
	}
}

File: Applications/Desktop/WpfMarkupTest/DomainApp/Models/VisitorModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DomainApp.Models
{
    public class VisitorModel
    {
        Document<Visitor> store = Document.Open<Visitor>("visitors.xml");

        public Visitor[] ReadVisitors()
        {
            return store.ToArray();
        }

        public void WriteVisitor(string name)
        {
            Visitor visitor = store.Find(v => v.Name == name);
            if (visitor == null)
                store.Add(new Visitor { Name = name });
            else
                visitor.Revisit();
            store.Save();
        }
    }
}
